src/nrrd/formatNRRD.c

```diff
@@ -522,7 +522,12 @@ _nrrdFormatNRRD_read(FILE *file, Nrrd *nrrd, NrrdIoState *nio) {
   /* NOTE: you have to open dataFile even in the case of skipData, because
      caller might have set keepNrrdDataFileOpen, in which case you need to
      do any line or byte skipping if it is specified */
-  valsPerPiece = nrrdElementNumber(nrrd)/_nrrdDataFNNumber(nio);
+  if (nio->numberOfElementsPerChunk != 0) {
+    valsPerPiece = nio->numberOfElementsPerChunk/_nrrdDataFNNumber(nio);
```

Comment:
What happens if this is not an integer?

Reply:
I didn't think I could be not integer, I changed the condition > 0.001